src/aspire/line/line.py

```diff
@@ -0,0 +1,138 @@
+        """
+        Initialize a Line Object. Change later (similar intuition from Image class)
+
+        :param data: Numpy array containing image data with shape
+            `(..., resolution, resolution)`.
+        :param dtype: Optionally cast `data` to this dtype.
+            Defaults to `data.dtype`.
+        """
+        self.dtype = np.dtype(dtype)
+        if data.ndim == 2:
+            data = data[np.newaxis, :, :]
+        if data.ndim < 3:
+            assert "Projection Dimensions should be more than Three-Dimensions"
+        self._data = data.astype(self.dtype, copy=False)
+        self.ndim = self._data.ndim
+        self.shape = self._data.shape
+        self.stack_shape = self._data.shape[:-2]
+        self.stack_n_dim = self._data.ndim - 2
+        self.n = np.product(self.stack_shape)
+        self.n_angles = self._data.shape[-2]
+        self.n_radial_points = self._data.shape[-1]
+
+        # Numpy interop
+        # https://numpy.org/devdocs/user/basics.interoperability.html#the-array-interface-protocol
+        self.__array_interface__ = self._data.__array_interface__
+        self.__array__ = self._data
+
+    def _check_key_dims(self, key):
+        if isinstance(key, tuple) and (len(key) > self._data.ndim):
+            raise ValueError(
+                f"Line stack_dim is {self.stack_n_dim}, slice length must be =< {self.n_dim}"
+            )
+
+    def __getitem__(self, key):
+        self._check_key_dims(key)
+        return self.__class__(self._data[key])
+
+    def __setitem__(self, key, value):
+        self._check_key_dims(key)
+        self._data[key] = value
+
+    def stack_reshape(self, *args):
+        """
+        Reshape the stack axis.
+
+        :*args: Integer(s) or tuple describing the intended shape.
+
+        :returns: Line instance
+        """
+
+        # If we're passed a tuple, use that
+        if len(args) == 1 and isinstance(args[0], tuple):
+            shape = args[0]
+        else:
+            # Otherwise use the variadic args
+            shape = args
+
+        # Sanity check the size
+        if shape != (-1,) and np.prod(shape) != self.n:
+            raise ValueError(
+                f"Number of sinogram images {self.n_images} cannot be reshaped to {shape}."
+            )
+
+        return self.__class__(self._data.reshape(*shape, *self._data.shape[-2:]))
+
+    def asnumpy(self):
+        """
+        Return image data as a (<stack>, angles, radians)
+        read-only array view.
+
+        :return: read-only ndarray view
+        """
+
+        view = self._data.view()
+        view.flags.writeable = False
+        return view
+
+    def copy(self):
+        return self.__class__(self._data.copy())
+
+    def __str__(self):
+        return f"Line(n_images = {self.n}, n_angles = {self.n_points}, n_radial_points = {self.n_radial_points})"
+
+    @property
+    def stack(self):
+        return self.n_images
+
+    def back_project(self, angles):
```

Comment:
Maybe we should discuss if we want to call this `backproject` like we do for the `Image` class.

Reply:
Yes, it should be called `backproject`